src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -51,6 +60,9 @@ public NuGetSdkResolver()
         internal NuGetSdkResolver(IGlobalJsonReader globalJsonReader)
         {
             _globalJsonReader = globalJsonReader;
+
+            _globalJsonReader.FileReadStart += (_, args) => NuGetSdkResolverEventSource.Instance.GlobalJsonReadStart(args.FullPath, args.ProjectFullPath, args.SolutionFullPath);
+            _globalJsonReader.FileReadStop += (_, args) => NuGetSdkResolverEventSource.Instance.GlobalJsonReadStop(args.FullPath, args.ProjectFullPath, args.SolutionFullPath);
```

Comment:
If we need the events, then these lambda functions should be converted to named methods, and then unsubscribed from the events in a Dispose method.

Reply:
The SDK resolver is never disposed of and is expected to be cleaned up along with the parent process when it exits.  But I could move these calls to the `GlobalJsonReader` itself, I was just trying to separate concerns and not sprinkle event tracing all over the place.  But that would keep me from having to introduce the new `GlobalJsonFileReadEventArgs` type.

Reply:
Unsubscribing from events is best practise. We've had [memory leaks in the past](https://github.com/NuGet/Home/issues/8471), and may well again, so people in the team should be calling what looks like bugs. If this class has special justification why it shouldn't follow best practises, it should be documented with comments. Not everyone is familar with SDK resolvers, and their lifecycle, and once this PR is merged, they won't have the context of this PR's comments. The PR description didn't explain this object lifecycle, but even if it did, once the PR is merged that context gets lost.

However, unless following the well known pattern will cause problems, I'd still suggest implementing the best practise, so that if static code analysis tools are run, they don't get tripped up and force a person to then investigate and find the comments.